### PR TITLE
Data preserve

### PR DESCRIPTION
**Pull request by Spacelama587**
Please do not invalidate the pull request for doing pair programing activity alone.
I have attached email screenshot from student success below

Feature implemented : Preserved Form Data

When the user changes the content of any input field, the data is saved to the local storage.
When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

**Comment by Spacelama587** [comment]
EmailSS

File: script.js [hunk 1]
```diff
@@ -621,4 +621,18 @@ window.onload = function () {
   }
 
   button7.addEventListener('click', openModalWithProject7);
-};
\ No newline at end of file
+};
+
```
Comment (topeogunleye) [inline script.js:625]: Hello @Spacelama587,

Thank you for submitting your project. However, I noticed that your project does not meet any of the requirements listed in the instructions. Please review the instructions carefully and make the necessary changes before resubmitting your project. Here are some specific points that you need to address:

- Your contact form information is not persistent to page reloads. You need to use the localStorage API to save and retrieve the form data in the browser.
- You did not store the form data for all inputs in a single object, nor did you save it to a single local storage registry. You need to create an object with properties for each input value and use the localStorage.setItem method to save it as a string.
- You did not update the data object and save it to the local storage when the user changes the content of any input field in the form. You need to add event listeners to each input field and use the localStorage.setItem method to update the stored data.
- You did not retrieve the data object from the local storage and pre-fill the input fields with the data when the user loads the page. You need to use the localStorage.getItem method to get the stored data as a string and use the JSON.parse method to convert it back to an object. Then you need to use a window.onload event handler to populate the input values with the data object properties.

Please make these changes and resubmit your project for another review. If you have any questions or need any help, please reach out to me or your peers on Slack.

Happy coding!

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Spacelama587, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `data-preserve` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-data-preserve`. This makes it easier to track the changes and the purpose of each branch.
- Kindly change the name of your project from Hello-Microverse to portfolio as this is not an hello microverse project.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Spacelama587** [comment]
Please review this:  https://github.com/Spacelama587/Portfolio-Hero/pull/13

File: script.js
Comment (wayungi) [inline script.js:1]: - [ ] You have not used localStoarge at all to preserve data in your form. Kindly for the instructions below and resubmit for re-view once your project is done. 
reqs

**Review by wayungi — COMMENTED** [review]
Hi @Spacelama587 , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] Fair attemp at preserving data in local storage 😓  


 ## Required Changes ♻️
_Check the comments under the review._
- [ ] Please use local storage to preserve data in the browser. The instructions are below.
reqs

Below is a screen shot of your form showing data in the form but nothing in local storage
store


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wayungi** [comment]
> Please review this: #13

Hello @Spacelama587 , You are supposed to make a PR from the branch hat has the changes to be reviewed. If the changes are in PR#13, please merge it to the current branch and re-submit .

**Pull request merged**